GCCcore/9.3.0 fails to find GMP in stage2 for ISL on OpenSUSE Leap 15.2

After fixing binutils on OpenSUSE Leap 15.2 I'm trying to build the full foss/2020a toolchain.

During the build of GCCcore/9.3.0 while configuring ISL for stage 2 with the following `run_cmd` command, it finds the header but not the static library in the stage2 prefix:
```
./configure --prefix=/tmp/eb-build/GCCcore/9.3.0/system-system/gcc-9.3.0/stage2_stuff --with-pic --disable-shared --with-gmp=system --with-gmp
-prefix=/tmp/eb-build/GCCcore/9.3.0/system-system/gcc-9.3.0/stage2_stuff  --build=x86_64-pc-linux-gnu --host=x86_64-pc-linux-gnu
```
```
checking which gmp to use... system
checking gmp.h usability... yes
checking gmp.h presence... yes
checking for gmp.h... yes
checking for main in -lgmp... no
configure: error: gmp library not found
```
The failing `config.log` test uses this command line:
```
configure:19076: gcc -o conftest -O3 -fomit-frame-pointer -malign-double -fstrict-aliasing -ffast-math -I/tmp/eb-build/GCCcore/9.3.0/system-system/gcc-9.3.0/stage2_stuff/include  -L/tmp/eb-build/GCCcore/9.3.0/system-system/gcc-9.3.0/stage2_stuff/lib -I/tmp/eb-build/GCCcore/9.3.0/system-system/gcc-9.3.0/stage2_stuff/include  -L/tmp/eb-build/GCCcore/9.3.0/system-system/gcc-9.3.0/stage2_stuff/lib  conftest.c -lgmp  -lgmp  >&5 
```

The given `-L/tmp/eb-build/GCCcore/9.3.0/system-system/gcc-9.3.0/stage2_stuff/lib` location for GMP is incorrect as on this platform, the library is in a `lib64` directory: `/tmp/eb-build/GCCcore/9.3.0/system-system/gcc-9.3.0/stage2_stuff/lib64/libgmp.a`

@bartoldeman found a part in `gcc.py` that assumes that libraries are installed in a `lib` directory, not sure if that's a root cause or just an attempt to plaster over this GCC build behaviour on some platforms: https://github.com/easybuilders/easybuild-easyblocks/blob/6e63e1498a849f5e0bc8c57eef614e3bd4fea8ff/easybuild/easyblocks/g/gcc.py#L595-L601

--- a/easybuild/easyblocks/g/gcc.py
+++ b/easybuild/easyblocks/g/gcc.py
@@ -599,6 +599,14 @@ class EB_GCC(ConfigureMake):
                     if lib == "gmp":
                         # make sure correct GMP is found
                         libpath = os.path.join(stage2prefix, 'lib')
+
+                        # fall back to lib64 directory if lib was not found,
+                        # give up if neither are there
+                        if not os.path.exists(libpath):
+                            libpath += '64'
+                        if not os.path.exists(libpath):
+                            raise EasyBuildError("lib(64) subdirectory not found in %s!" % stage2prefix)
+
                         incpath = os.path.join(stage2prefix, 'include')
 
                         cppflags = os.getenv('CPPFLAGS', '')

